discovery/kubernetes_api_discovery.go

```diff
@@ -56,16 +56,16 @@ func (k *K8sAPIDiscoverer) servicesForNode(hostname, ip string) []service.Servic
 		for _, pod := range podList {
 			// We require an annotation called 'ServiceName' to make sure this is
 			// a service we want to announce.
-			if pod.ServiceName() == "" {
+			if len(pod.ServiceName()) < 1 {
```

Comment:
Comparing a string to `""` is much more expensive than comparing length of the string. It turns out, from the profiler, that this was one of the most expensive calls in the whole service. This eliminates that call entirely.

Reply:
This might've been a profiler fluke. Both approaches should produce the same instructions according to this thread: https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go

Reply:
Would be a cool compiler bug report if, indeed, it's unable to optimise correctly in this particular piece of code :)